vulnerabilities/importers/gitlab.py

```diff
@@ -0,0 +1,347 @@
+        )
+
+
+def parse_yaml_file(file):
+    """
+    Take file name as input and parse the yaml file
+    to get AdvisoryData object
+
+    Sample YAML file:
+    ---
+    identifier: "GMS-2018-26"
+    package_slug: "packagist/amphp/http"
+    title: "Incorrect header injection check"
+    description: "amphp/http isn't properly protected against HTTP header injection."
+    pubdate: "2018-03-15"
+    affected_range: "<1.0.1"
+    fixed_versions:
+    - "v1.0.1"
+    urls:
+    - "https://github.com/amphp/http/pull/4"
+    cwe_ids:
+    - "CWE-1035"
+    - "CWE-937"
+    identifiers:
+    - "GMS-2018-26"
+    """
+    with open(file, "r") as f:
+        gitlab_advisory = yaml.safe_load(f)
+    if not isinstance(gitlab_advisory, dict):
+        logger.error(f"parse_yaml_file: yaml_file is not of type `dict`: {gitlab_advisory!r}")
+        return
+
+    # refer to schema here https://gitlab.com/gitlab-org/advisories-community/-/blob/main/ci/schema/schema.json
+    aliases = gitlab_advisory.get("identifiers")
+    summary = ". ".join([gitlab_advisory.get("title"), gitlab_advisory.get("description")])
+    urls = gitlab_advisory.get("urls")
+    references = [Reference.from_url(u) for u in urls]
+    date_published = dateparser.parse(gitlab_advisory.get("pubdate"))
+    date_published = pytz.utc.localize(date_published)
+    fixed_versions = gitlab_advisory.get("fixed_versions")
+    affected_version_range = None
+    affected_range = gitlab_advisory.get("affected_range")
+    purl: PackageURL = get_purl(gitlab_advisory.get("package_slug"))
+    if not purl:
+        logger.error(f"parse_yaml_file: purl is not valid: {file!r}")
+        return AdvisoryData(
+            aliases=aliases,
+            summary=summary,
+            references=references,
+            date_published=date_published,
+        )
+    vrc: VersionRange = RANGE_CLASS_BY_SCHEMES[purl.type]
+    version_class = vrc.version_class
+    gitlab_native_schemes = ["pypi", "gem", "npm", "go", "packagist"]
+    gitlab_scheme = GITLAB_SCHEME_BY_PURL_TYPE[purl.type]
+    try:
+        if gitlab_scheme in gitlab_native_schemes:
+            if affected_range:
+                affected_version_range = from_gitlab_native(
+                    gitlab_scheme=gitlab_scheme, string=affected_range
+                )
+        else:
+            affected_version_range = vrc.from_native(affected_range) if affected_range else None
+    except Exception as e:
+        logger.error(
+            f"parse_yaml_file: affected_range is not parsable: {affected_range!r} type:{purl.type} {e} {traceback.format_exc()}"
+        )
+
+    parsed_fixed_versions = []
+    for fixed_version in fixed_versions or []:
+        try:
+            fixed_version = version_class(fixed_version)
+            parsed_fixed_versions.append(fixed_version)
+        except Exception as e:
+            logger.error(f"parse_yaml_file: fixed_version is not parsable`: {fixed_version!r}")
```

Comment:
Add the stacktrace to the log... otherwise this is not easy to diagnose.
```suggestion
            logger.error(f"parse_yaml_file: fixed_version is not parsable`: {fixed_version!r} error: {e!r}\n {traceback.format_exc()}")
```